les/server.go

```diff
@@ -205,14 +206,14 @@ func (s *LesServer) Start(srvr *p2p.Server) {
 		}
 	}
 
-	maxCapacity := s.freeClientCap * uint64(s.maxPeers)
-	if totalRecharge > maxCapacity {
-		maxCapacity = totalRecharge
+	s.maxCapacity = s.freeClientCap * uint64(s.maxPeers)
+	if totalRecharge > s.maxCapacity {
+		s.maxCapacity = totalRecharge
 	}
-	s.fcManager.SetCapacityLimits(s.freeClientCap, maxCapacity, s.freeClientCap*2)
-	s.freeClientPool = newFreeClientPool(s.chainDb, s.freeClientCap, 10000, mclock.System{}, func(id string) { go s.protocolManager.removePeer(id) })
-	s.protocolManager.peers.notify(s.freeClientPool)
-
+	s.fcManager.SetCapacityLimits(s.freeClientCap, s.maxCapacity, s.freeClientCap*2)
+	s.clientPool = newClientPool(s.chainDb, s.freeClientCap, 10000, mclock.System{}, func(id enode.ID) { go s.protocolManager.removePeer(peerIdToString(id)) })
+	s.clientPool.setPriceFactors(priceFactors{0, 1, 1}, priceFactors{0, 1, 1})
```

Comment:
I think the time cost is a bit too high.

E.g. for a paid client(cap=1000,000). If the client connects 1s, the time cost is 10^9. While for a normal request cost is much lower.

Besides, we will also consider time cost for `estimated balance`. The refresh interval in `lazyQueue` is 10s.
It means the estimated balance will much much higher than the actual balance. It can result in we always need to dump all items from estimated queue and then push them to pop queue to pick item.

Reply:
Currently I think there is probably no "theoretically optimal" value for the ratio of time and request costs and the current default value is just a wild guess but I think it is in a reasonable range. As you wrote, 1s of connection with 1M capcacity costs 10^9 units. 1M capacity means that 10^9 request cost per second is guaranteed so if you use your guaranteed amount then the request cost will be equal to the time cost. If only a few clients are requesting then they also get extra buffer recharge which allows for up 10 times higher request rate so if the client is sucking data at maximum rate then the request cost can be even 10 times higher than the time cost.

Reply:
Okay. Can we add a metric here to collect the average number of items be popped for `MultiPop` function? I have a bad feeling that we will pop the whole queue.